parsl/tests/test_monitoring/test_exit_helper.py

```diff
@@ -0,0 +1,56 @@
+import multiprocessing
+import signal
+
+import psutil
+import pytest
+
+from parsl.monitoring.monitoring import join_terminate_close_proc
+from parsl.multiprocessing import ForkProcess
+
+
+def noop():
+    pass
+
+
+@pytest.mark.local
+def test_pre_exit():
+    """Test calling against a process that has already exited."""
```

Comment:
FWIW, you might consider using the test name itself as the documentation.  Consider:
```python
@pytest.mark.local
def test_join_helper_returns_on_already_finished_process():
```
Now the intention is clear in the test output, as for example when one runs `pytest` with arguments like `-vsx`.

Not a blocker.